backend/dataall/modules/datasets/services/dataset_location_service.py

```diff
@@ -87,14 +87,8 @@ def remove_storage_location(uri: str = None):
         with get_context().db_engine.scoped_session() as session:
             location = DatasetLocationRepository.get_location_by_uri(session, uri)
             dataset = DatasetRepository.get_dataset_by_uri(session, location.datasetUri)
-            has_shares = ShareObjectRepository.has_shared_items(session, location.locationUri)
-            if has_shares:
-                raise ResourceShared(
-                    action=DELETE_DATASET_FOLDER,
-                    message='Revoke all folder shares before deletion',
-                )
-
-            ShareObjectRepository.delete_shares(session, location.locationUri)
+            DatasetService.check_before_delete(session, location.locationUri, action=DELETE_DATASET_FOLDER)
```

Comment:
We can check this on the line 89 before getting dataset. Search for dataset is unnecessary, if the check fails.

Reply:
Resolved in https://github.com/data-dot-all/dataall/pull/1213/commits/ebdcdb058b961c2de32837de52268554e16f8680